### ISSUE TITLE
Support sharing encrypted realms across processes

### ISSUE THREAD

**Issue by jpsim** [realm/realm-cocoa#1693]

**Comment by asjesset** [comment]
Any progress on this? We are wanting to make an iMessage/home screen extension and need this functionality.

**Comment by hdoria** [comment]
Any progress on this?

**Comment by beseder42** [comment]
We are also waiting for this! What it the time horizon to see this in a release?

**Comment by jbouaziz** [comment]
We also need that functionality, were you able to make it work?

**Comment by emlynmac** [comment]
I'm keen to get this functionality too!

**Comment by jongarate** [comment]
This is a major pain in the ass these days. Would be SO welcome.

**Comment by florianwalter** [comment]
Maybe i am missing something here, but e.g. with the latest RealmSwift 3.5.0, this works.

**Comment by cagricolak** [comment]
nowadays the problem still continues, 2019. thanks money.

**Comment by thecoolwinter** [comment]
Really need this functionality, would appreciate an update soon!

**Comment by cguzmanceti** [comment]
Hi. Any progress with this issue?? I need access to encrypted realm DB from a notification service extension, but I cann't thank you this issue 😕. Is any release planned that includes the correction of this issue? Thank you so much!

**Comment by bmunkholm** [comment]
Hi @cguzmanceti, 
Unfortunately, this is not one of the features that we are prioritizing near term. You can see our current priorities [here](https://www.mongodb.com/realm#roadmap).

**Comment by cagricolak** [comment]
5 years! what happened !

**Comment by cguzmanceti** [comment]
Hi @cagricolak ,
What happening???

**Comment by caravaneer404** [comment]
I still waiting 🐸

**Comment by XabierGoros** [comment]
It would be great to have this feature to be honest.

**Comment by yxzhaoo** [comment]
Hi guys, is there any work around for this?

**Comment by thecoolwinter** [comment]
This feature is more relevant now with Widgets on Apple devices, and not being able to access encrypted realms from them. Would be awesome to have.

**Comment by si-hartmann** [comment]
This is now a dramatic show stopper as all(!) app data can now be accessed when an app is installed on a Silicon Mac. Apps can even be installed and used on a M1 Mac without(!) installing them from the Apple App Store. This means that any data that is not encrypted can be read and exported from any app! That is a huge security vulnerability.

Are there any plans to get this feature implemented after 5,5 years?

**Comment by leemaguire** [comment]
@si-hartmann what you are saying does not look related to multiprocess support, but rather the sandboxing mechanism that iOS uses. Could you clarify your use case? 

Also relying on iOS sandboxing for security is not recommended, as jailbreaking & side loading apps with system integrity protection turned off (on an M1 Mac) will let anyone view the filesystem of the app.

**Comment by si-hartmann** [comment]
@leemaguire Total agree with your second paragraph about security. But: Not being able to use encrypted realms in extensions forces apps to either use unencrypted realms or to use field level encryption - a big step and probably not down often. 

Or we do not use extensions on iOS. But do you expect your iOS developers to not provide any iOS extension? The eco system does not work like that. Apple is relaying heavily on extensions for a nice UX. So it is Realm, that does not provide the necessary realm level encryption for developing these extensions in a secure way. All tough technical being possible now.

Sure jailbreaking was always a possible attack against iOS sandboxing. But now everybody(!) can load any app on the M1 Mac and then read (and edit!!) all data that the app uses. No jailbreak or anything else needed. I guess this security issue will become visible and huge soon. Should be addressed very soon!

**Comment by XabierGoros** [comment]
> Also relying on iOS sandboxing for security is not recommended, as jailbreaking & side loading apps with system integrity protection turned off (on an M1 Mac) will let anyone view the filesystem of the app.

That's absolutely correct @leemaguire, and that's the main reason of why we need this feature to be developed ASAP.

**Comment by gixdev** [comment]
Any progress? We need it badly

**Comment by RomainPct** [comment]
Still no news on the topic ? There is no clean workaround to use an encrypted realm in an iOS extension as we can't be sure to kill the other processes in app.
- Will it be part of a future release ?
- Is there any known work around to solve it while waiting for encrypted realm support across processes ?

**Comment by tgoyne** [comment]
Yes, this was released in 10.38.0.
